ragna/core/_components.py

```diff
@@ -76,6 +89,29 @@ def _protocol_model(cls) -> Type[pydantic.BaseModel]:
         return merge_models(cls.display_name(), *cls._protocol_models().values())
 
 
+# Just for demo purposes. We need to move the actual class here.
+# See https://github.com/Quansight/ragna/pull/354#discussion_r1526235318
+class Embedding:
+    embedding: list[float]
+    chunk: Chunk
+
+    def __init__(self, embedding: list[float], chunk: Chunk):
+        super().__init__()
+        self.embedding = embedding
+        self.chunk = chunk
+
+
+class GenericEmbeddingModel(Component, ABC):
+    _EMBEDDING_DIMENSIONS: int
+
+    @abstractmethod
+    def embed_chunks(self, chunks: list[Chunk]) -> list[Embedding]:
```

Comment:
If we actually don't need the `embed_text` as I reckon below, we can probably name this just

```suggestion
    def embed(self, chunks: list[Chunk]) -> list[Embedding]:
```

?

Reply:
If we don't separate out the embed_text function, then you need to create a chunk holding the prompt in order to query the database.  I dislike this, so I would prefer to keep them separated.

Reply:
Right, makes sense.